Engine/Models.cs

```diff
@@ -0,0 +1,61 @@
+namespace Engine;
+
+[Serializable]
+public class Settings
+{
+    public int MaxHandCards { get; set; } = 5;
+    public int? RandomSeed { get; set; }
+}
+
+[Serializable]
+public class GameState
+{
+    public Settings? Settings { get; set; }
+    public List<Player> Players { get; set; }
+    public List<List<Card>> CenterPiles { get; set; }
```

Comment:
I would usually recommend a seperate class to avoid Double nested list. E.g. Make a `CenterPile` class which is a `List<Card>` and then have `List<CenterPile> CenterPiles` here